src/main/java/com/google/sps/DataServlet.java

```diff
@@ -153,8 +162,35 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     if (mutationNumber > currDataGraph.numMutations()) {
       diff = Utility.getMultiMutationAtIndex(mutList, mutationNumber);
     }
+    List<String> nodeNames = new ArrayList<>();
+    try {
+      JsonArray nodeNameArr = JsonParser.parseString(nodeNamesParam).getAsJsonArray();
+      for (int i = 0; i < nodeNameArr.size(); i++) {
+        String curr = nodeNameArr.get(i).getAsString().trim();
+        if (curr.length() > 0) {
+          nodeNames.add(curr);
+        }
+      }
+    } catch (JsonSyntaxException | IllegalStateException e) {
```

Comment:
Does this mean the input is invalid? If so, you should probably propagate the error to the user.